lib/srv/db/mongodb/connect.go

```diff
@@ -153,6 +161,43 @@ func (e *Engine) getConnectionOptions(ctx context.Context, sessionCtx *common.Se
 	}, nil
 }
 
+func (e *Engine) getAuthenticator(ctx context.Context, sessionCtx *common.Session) (authenticator auth.Authenticator, err error) {
+	// Currently, the MongoDB Atlas IAM Authentication doesn't work with IAM
+	// users. Here we provide a better error message to the users.
+	if sessionCtx.Database.GetType() == types.DatabaseTypeMongoAtlas && awsutils.IsUserARN(sessionCtx.DatabaseUser) {
+		return nil, trace.BadParameter("MongoDB Atlas AWS IAM Authentication with IAM users is not supported.")
+	}
+
+	switch {
+	case sessionCtx.Database.GetType() == types.DatabaseTypeMongoAtlas && awsutils.IsRoleARN(sessionCtx.DatabaseUser):
```

Comment:
the `sessionCtx.Database.GetType() == types.DatabaseTypeMongoAtlas` check is duplicated can we assign the result to local variable and reuse it ?